pkg/lease/model.go

```diff
@@ -11,19 +11,21 @@ import (
-	ID                       *string                `json:"id,omitempty" dynamodbav:"Id,omitempty"`                                             // Lease ID
-	Status                   *Status                `json:"leaseStatus,omitempty" dynamodbav:"LeaseStatus,omitempty"`                           // Status of the Lease
-	StatusReason             *StatusReason          `json:"leaseStatusReason,omitempty" dynamodbav:"LeaseStatusReason,omitempty"`               // Reason for the status of the lease
-	CreatedOn                *int64                 `json:"createdOn,omitempty" dynamodbav:"CreatedOn,omitempty"`                               // Created Epoch Timestamp
-	LastModifiedOn           *int64                 `json:"lastModifiedOn,omitempty" dynamodbav:"LastModifiedOn,omitempty"`                     // Last Modified Epoch Timestamp
-	BudgetAmount             *float64               `json:"budgetAmount,omitempty" dynamodbav:"BudgetAmount,omitempty"`                         // Budget Amount allocated for this lease
-	BudgetCurrency           *string                `json:"budgetCurrency,omitempty" dynamodbav:"BudgetCurrency,omitempty"`                     // Budget currency
-	BudgetNotificationEmails *[]string              `json:"budgetNotificationEmails,omitempty" dynamodbav:"BudgetNotificationEmails,omitempty"` // Budget notification emails
-	StatusModifiedOn         *int64                 `json:"leaseStatusModifiedOn,omitempty" dynamodbav:"LeaseStatusModifiedOn,omitempty"`       // Last Modified Epoch Timestamp
-	ExpiresOn                *int64                 `json:"expiresOn,omitempty" dynamodbav:"ExpiresOn,omitempty"`                               // Lease expiration time as Epoch
-	Metadata                 map[string]interface{} `json:"metadata,omitempty" dynamodbav:"Metadata,omitempty"`                                 // Arbitrary key-value metadata to store with lease object
+	AccountID                *string       `json:"accountId,omitempty" dynamodbav:"AccountId" schema:"accountId,omitempty"`                                                        // AWS Account ID
+	PrincipalID              *string       `json:"principalId,omitempty" dynamodbav:"PrincipalId" schema:"principalId,omitempty"`                                                  // Azure User Principal ID
+	ID                       *string       `json:"id,omitempty" dynamodbav:"Id,omitempty" schema:"id,omitempty"`                                                                   // Lease ID
+	Status                   *Status       `json:"leaseStatus,omitempty" dynamodbav:"LeaseStatus,omitempty" schema:"status,omitempty"`                                             // Status of the Lease
+	StatusReason             *StatusReason `json:"leaseStatusReason,omitempty" dynamodbav:"LeaseStatusReason,omitempty" schema:"leaseStatusReason,omitempty"`                      // Reason for the status of the lease
```

Comment:
Having `schema:"leaseStatusReason,omitempty"` means someone can pass in this parameter and it will get parsed and put into the query.  I don't see this in the swagger doc.

Reply:
I updated schema:"-" for LeaseStatusReason